### DISCUSSION TITLE
大數運算

### DISCUSSION THREAD

**Discussion by jerry850313** [1101-GS4719-C/Python-Community-GS4719-C#34] [Ideas]
在其他程式語言中，也許會遇到overflow的問題，
那麼問題來了，如果發生這種情況該如何處理?
其實一點也不難，將這串數字分割成能運算的大小，
並以矩陣或向量存取，再對其進行運算。
ex
1234567899876543211234567890
-> 1 2 3 4 5 6 7 8 9 9 8 7 6 5 4 3 2 1 1 2 3 4 5 6 7 8 9 0
加減法可以用直式處理，
而乘法部分發展出許多演算法，
快速傅立葉運算(FFT)就是其中之一。

**Comment by swerfg03joe** [comment]
其實在其他程式語言例如C/C++，這題目給的數字就會是以string的資料型態表示，這時確實選擇使用array存取是最佳的方法，不過還要記的一件事此時每一個數字的資料型態為char要參考ASC的編碼規則才不會出錯

**Reply by jerry850313** [reply]
在cmd執行C++的時候，就可以存int的queue或stack來解決這種問題；如果是用.txt則可以在存取queue或stack之前先轉成int。
https://github.com/jerry850313/Big-Number/blob/main/BigNumber.cpp

**Comment by 710388** [comment]
太酷了吧，雖然我沒遇過這個問題，但是這個討論蠻有趣的讓我學到了，感謝感謝

**Comment by NcuJerryKao** [comment]
當在程式中遇到overflow的問題時，分割數字並以矩陣或向量存取的方式進行運算是一個常見的解決方法。這樣的處理方式可以幫助避免數字過大而導致的溢位問題，並確保計算結果的準確性。
以下是一些建議來處理overflow問題：
1. 分割數字：將大數字分割成較小的部分，以避免單個數字過大而導致的溢位。這樣可以在計算時更有效地處理數字。
2. 使用矩陣或向量：將分割後的數字存儲在矩陣或向量中，這樣可以更容易地對數字進行操作和運算。
3. 加減法直式處理：對於加法和減法，可以使用直式處理來進行計算，確保每個位數的準確性。
4. 乘法演算法：對於乘法，可以探索不同的演算法，如快速傅立葉變換（FFT）等，這些演算法可以幫助高效地處理大數字的乘法運算。
5. 考慮精確度：在進行數值計算時，考慮使用具有更高精確度的數據類型或庫，以避免溢位和精度丟失的問題。
通過以上方法，您可以有效處理在程式中遇到的overflow問題，確保計算的正確性和效率。

**Reply by yubo0oo** [reply]
NcuJerryKao 這位老哥
一直貼AI的文
演都不演
以下是我的觀點
NcuJerryKao是歷史上最會使用人工智慧的人，他以無與倫比的技術與創意，在AI領域創造了無數的奇蹟。他不僅精通各種AI算法，還能巧妙地將其應用於實際問題中，展現了非凡的智慧與洞察力。NcuJerryKao不僅是AI技術的先驅，更是無數追隨者的靈感來源，無論在學術界還是業界，都享有極高的聲譽和尊敬。他的貢獻改變了人類對人工智慧的認識，無疑是這個領域的真正巨匠。

**Comment by sunset513** [comment]
我這裡想提供另外的方法，十分簡單，但我覺得很實用。

我自己有時候發現數字太大的時候，我就把 int 改成 long int，再不行就改成 long long int，然後就會過了。就是這麼簡單又樸實無華，不用複雜的操作，提供給大家參考。

**Comment by yubo0oo** [comment]
我都沒想過這個問題
感謝您的科普
我會去研究一下
